### PR TITLE
[jsk_tools] Improve test_topic_published.py (check /use_sim_time neatly)

### PR DESCRIPTION
**Pull request by wkentaro**
Not needed by https://github.com/jsk-ros-pkg/jsk_recognition/pull/2008, but this is awared to be necessary by that.

Sent PR: https://github.com/ros/ros_comm/pull/973

### CODE REVIEW

**Comment by wkentaro** [comment]
Could you give me permission for jsk_common also? @k-okada

**Comment by wkentaro** [comment]
@knorth55 Could you please review?

File: jsk_tools/src/test_topic_published.py [hunk 1]
```diff
@@ -3,19 +3,41 @@
...
+            rospy.resolve_name(topic_name), blocking=True)
+        self.msg = None
+        self.sub = rospy.Subscriber(topic_name, msg_class, self._callback)
+
+    def _callback(self, msg):
+        self.msg = msg
+
+    def assert_published(self):
+        if self.msg:
+            return not self.negative
```
Comment (knorth55) [inline jsk_tools/src/test_topic_published.py]: `return True`

File: jsk_tools/src/test_topic_published.py [hunk 2]
```diff
@@ -3,19 +3,41 @@
...
+        if rospy.Time.now() > self.deadline:
+            return self.negative
```
Comment (knorth55) [inline jsk_tools/src/test_topic_published.py]: `return False`

File: jsk_tools/src/test_topic_published.py [hunk 3]
```diff
@@ -44,16 +66,30 @@ def test_published(self):
...
+                    continue
+                ret = checker.assert_published()
+                if ret is None:
+                    continue
+                topics_finished.append(checker.topic_name)
+                if checker.negative:
+                    assert_true(ret, 'Topic [%s] is published' %
+                                     (checker.topic_name))
+                else:
+                    assert_true(
```
Comment (knorth55) [inline jsk_tools/src/test_topic_published.py]: `assert_false(`

File: jsk_tools/src/test_topic_published.py [hunk 4]
```diff
@@ -3,19 +3,41 @@
...
+        self.deadline = rospy.Time.now() + rospy.Duration(timeout)
+        msg_class, _, _ = rostopic.get_topic_class(
+            rospy.resolve_name(topic_name), blocking=True)
+        self.msg = None
+        self.sub = rospy.Subscriber(topic_name, msg_class, self._callback)
+
+    def _callback(self, msg):
+        self.msg = msg
+
+    def assert_published(self):
```
Comment (knorth55) [inline jsk_tools/src/test_topic_published.py]: if you name `assert_published`, it is better and easier to understand to return `True` when msg is published and `False` when it isn't.

**Comment by wkentaro** [comment]
I think it is ok because it is set as `negative=True`, and following upstream code is better, don't you think?

**Comment by knorth55** [comment]
quite complicating for me.
Assert_published should return whether msg is published or not.
If negative is true, the return value of assert_published is opposite, and it is complicating

**Comment by wkentaro** [comment]
Ok, fixed.

**Comment by knorth55** [comment]
:+1:

**Comment by wkentaro** [comment]
@k-okada Could you please merge this?

**Pull request merged**